MapboxCoreNavigation/NavigationLocationManager.swift

```diff
@@ -9,38 +9,22 @@ import UIKit
 #endif
 
 /**
- `NavigationLocationManager` is the base location manager which handles
- permissions and background modes.
+ `NavigationLocationManager` is the base location manager which handles permissions and background modes.
+ 
+ If it's necessary to change the default behavior in `NavigationLocationManager`, you can subclass this class.
  */
 @objc(MBNavigationLocationManager)
 open class NavigationLocationManager: CLLocationManager {
     
     var lastKnownLocation: CLLocation?
     
-    /**
-     Indicates whether the location manager’s desired accuracy should update
-     when the battery state changes.
-     */
-    @available(*, deprecated, message: "Manually override desiredAccuracy instead.")
-    public var automaticallyUpdatesDesiredAccuracy = true
-    
-    var accuracyOverride: CLLocationAccuracy?
-    
-    override open var desiredAccuracy: CLLocationAccuracy {
-        get {
-            if let override = accuracyOverride { return override }
-            return  UIDevice.current.isPluggedIn ? kCLLocationAccuracyBestForNavigation : kCLLocationAccuracyBest
-        }
-        set {
-            accuracyOverride = newValue
-        }
-    }
-    
     override public init() {
         super.init()
         
         requestWhenInUseAuthorization()
         
+        desiredAccuracy = UIDevice.current.isPluggedIn ? kCLLocationAccuracyBestForNavigation : kCLLocationAccuracyBest
```

Comment:
This change assumes `kCLLocationAccuracyBestForNavigation` offers enough of an improvement over `kCLLocationAccuracyBest` that it’s worth enabling, even if there’s increased battery usage (as implied by the `isPluggedIn` check). This change reintroduces the use of `kCLLocationAccuracyBestForNavigation`, which we haven’t been using since #1206. So we need to test this change thoroughly before landing it.

Reply:
I'm okay with only using `kCLLocationAccuracyBest `